pkg/tools/gogen/gogen.go

```diff
@@ -0,0 +1,101 @@
+package gogen
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"strings"
+
+	"kusionstack.io/kclvm-go/pkg/logger"
+	"kusionstack.io/kclvm-go/pkg/spec/gpyrpc"
+)
+
+type GenStruct map[StructName][]*Field
+
+type StructName string
+
+type Field struct {
+	Name       string
+	SchemaType string
+	OmitEmpty  bool
+	Tag        string
+}
+
+func parseKclType(ktList []*gpyrpc.KclType) *GenStruct {
+	var schemaName StructName
+	genStruct := make(GenStruct, 0)
+	for _, kt := range ktList {
+		if kt.Type == "schema" {
+			schemaName = StructName(kt.SchemaName)
+			var schemaList []*Field
+			for k, p := range kt.Properties {
+				var field Field
+				field.Name = strings.ToUpper(k[:1]) + k[1:]
+				switch p.Type {
+				case "schema":
+					field.SchemaType = "*" + p.SchemaName
```

Comment:
For the `field.SchemaType`, how about using the code defined here: https://github.com/KusionStack/kclvm-go/blob/main/pkg/tools/gen/gengo.go#L109

Reply:
ok，got it. I find that you have commited the gengo code. My branch was behind from the commit code. I will use the code `GetTypeName`.